zcfan.c

```diff
@@ -228,14 +228,21 @@ static void get_config(void) {
         long pos = ftell(f);
         int ch;
         expect(pos >= 0);
-        fscanf_threshold(f, pos, "max_temp", FAN_MAX);
-        fscanf_threshold(f, pos, "med_temp", FAN_MED);
-        fscanf_threshold(f, pos, "low_temp", FAN_LOW);
+        fscanf_int_for_key(f, pos, "max_temp", rules[FAN_MAX].threshold);
+        fscanf_int_for_key(f, pos, "med_temp", rules[FAN_MED].threshold);
+        fscanf_int_for_key(f, pos, "low_temp", rules[FAN_LOW].threshold);
+        fscanf_int_for_key(f, pos, "watchdog_secs", watchdog_secs);
         if (ftell(f) == pos) {
             while ((ch = fgetc(f)) != EOF && ch != '\n') {}
         }
     }
 
+    /* Maximum value handled by the kernel is 120, and
+     * (watchdog_secs - WATCHDOG_GRACE_PERIOD_SECS) must stay positive. */
+    if (watchdog_secs < WATCHDOG_GRACE_PERIOD_SECS || watchdog_secs > 120) {
```

Comment:
Thanks, this is a good idea. That said, this is only read on startup, so I'm more inclined to conclusively tell the user that they have provided an invalid config and immediately exit with an error.

Reply:
Done, and I got annoyed by writing 120 many many times, so I added a cpp constant for it, but the string substitution got a bit ugly, just tell me if you prefer either:

  - 120 appearing many times through the file
  - replacing sizeof(...) complication with simply 3.